src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -391,7 +391,8 @@ function checkSharedValueUsage(
 }
 
 export function useAnimatedStyle<T extends AnimatedStyle>(
-  updater: BasicWorkletFunction<T>,
+  // animated style cannot be an array
+  updater: BasicWorkletFunction<T extends Array<unknown> ? never : T>,
```

Comment:
```suggestion
  updater: BasicWorkletFunction<Exclude<T, any[]>>,
```
Seems to be a better for readability.